js/main_pg_preload.js

```diff
@@ -1,29 +1,8 @@
 /* 
-
 Javascript for Preloader Animation and Page navigation
-
 */
 
+import Utils from "./utils";
 
-function delay(url) {
-    window.setTimeout( () =>{
-    	window.location = url 
-    }, 1200);
-}
-
-
-// Check the body content at this point and uncover preloader, check all resources are loaded
-
-
-let document_state = setInterval(function () {
-	if(document.readyState === "complete") {
-		clearInterval(document_state);
-		let body_tag = document.getElementById("main-body");
-		body_tag.className += " loaded";
-
-		window.setTimeout(() => {
-			let element = document.getElementById("preloader");
-			element.style.display = "none";
-		}, 6000);
-	}
-}, 10);

```

Comment:
* The shared logic inside `main_pg_preload.js` and `preload.js` can be extracted into another class, for example something we can call `utils`. We can then use the same function `preloadWithInterval` in both 'main_pg_preload.js' and 'preload.js'.
* If we have two JS files doing basically the same thing, we can just create one class with a function that does all the preloading logic. No need to have two separate JS files in this case. `main_pg_preload.js` and `preload.js`can be replaced with just one class.